### PR TITLE
Total power filter

### PR DESCRIPTION
**Pull request by nmahesh1412**
Sorry did this in the wrong branch.

### CODE REVIEW

**Comment by steven-murray** [comment]
Looks like there's an error in one of the tests (you can see where in the Details in the checks). Let me know if you need help identifying it.

File: src/edges_analysis/analysis/sky_models.py [hunk 1]
```diff
@@ -91,7 +91,9 @@ def get_index(
         lon: [None, np.ndarray] = None,
         sky_model: [None, SkyModel] = None,
     ) -> np.ndarray:
-        return self.index
+        index = np.zeros(len(lat))
```
Comment (steven-murray) [inline src/edges_analysis/analysis/sky_models.py:94]: Can you put a comment on top of this about why you do this? It's not obvious, and I might at some point go and revert it unless there's a comment why I shouldn't :-)

File: src/edges_analysis/analysis/beams.py [hunk 2]
```diff
@@ -523,12 +521,12 @@ def simulate_spectra(
...
 
     antenna_temperature_above_horizon = np.zeros((len(lst), len(beam.frequency)))
-    for i, j, temperature, _, _, _ in _iterate_through_lst_freq(
+    for i, j, temperature, convolution_product, ref_time, _ in _iterate_through_lst_freq(
         lst, ground_loss_file, beam, twenty_min_per_lst, sky_model, index_model, normalize_beam
     ):
         antenna_temperature_above_horizon[i, j] = temperature
 
-    return antenna_temperature_above_horizon, beam.frequency, lst
+    return antenna_temperature_above_horizon, beam.frequency, lst, convolution_product, ref_time
```
Comment (steven-murray) [inline src/edges_analysis/analysis/beams.py:529]: Adding extra outputs is causing one of the tests to fail. THat can be fixed by changing the call in the test easily enough, but I just wanted to make sure you really want to return these things -- what's the benefit of returning them?

File: src/edges_analysis/analysis/beams.py [hunk 3]
```diff
@@ -462,14 +457,17 @@ def _iterate_through_lst_freq(
...
+                antenna_temperature_above_horizon_list[i,j,horizon_mask]=(beam_above_horizon * sky_above_horizon[:,j])
             # Convolution between (beam at all frequencies) and (sky at all frequencies)
             yield (
                 i,
                 j,
                 np.nansum(beam_above_horizon * sky_above_horizon[:, j]) / n_pix_tot_no_nan,
-                sky_above_horizon,
-                beam_above_horizon,
+                antenna_temperature_above_horizon_list,
+                #sky_above_horizon,
```
Comment (steven-murray) [inline src/edges_analysis/analysis/beams.py:468]: can remove these commented lines

File: src/edges_analysis/analysis/beams.py [hunk 4]
```diff
@@ -419,7 +411,9 @@ def _iterate_through_lst_freq(
 
     # Advancing time (19:57 minutes UTC corresponds to 20 minutes LST)
     twenty_lst_min = dt.timedelta(minutes=19, seconds=57)
-
+    convol = []
+    antenna_temperature_above_horizon_list = np.zeros((len(lst), len(beam.frequency), len(sky_map[:,0])))
```
Comment (steven-murray) [inline src/edges_analysis/analysis/beams.py:415]: `len(sky_map[:,0])` can just be `len(sky_map)`.

**Review by steven-murray — COMMENTED** [review]
This looks mostly good, just want to make sure all of the changes are necessary and commented. Thanks @nmahesh1412

File: src/edges_analysis/analysis/beams.py (hunk 2 above)
Comment (nmahesh1412) [reply src/edges_analysis/analysis/beams.py:529]: These new outputs are needed for our LST binned analysis. Actually I am going to output one more product - sky above horizon list (values at all frequencies and lat/lon)

Comment (steven-murray) [reply src/edges_analysis/analysis/beams.py:529]: OK, excellent.

File: src/edges_analysis/analysis/sky_models.py [hunk 5]
```diff
@@ -91,6 +91,7 @@ def get_index(
         lon: [None, np.ndarray] = None,
         sky_model: [None, SkyModel] = None,
     ) -> np.ndarray:
+        ## to return an array of indices of len (latidute) like the other Index functions. This avoids the error in the beams.py code.
```
Comment (steven-murray) [inline src/edges_analysis/analysis/sky_models.py:94]: I think the neater way of doing this is `return np.ones_like(lat) * self.index`.

**Review by steven-murray — COMMENTED** [review]
There's a few files that need to be removed / ignored, and a couple of small comments.

File: src/edges_analysis/analysis/.ipynb_checkpoints/io-checkpoint.py [hunk 6]
```diff
@@ -0,0 +1,235 @@
+import h5py
```
Comment (steven-murray) [inline src/edges_analysis/analysis/.ipynb_checkpoints/io-checkpoint.py:1]: This file also should be removed

**Review by steven-murray — APPROVED** [review]
Looks good. Thanks @nmahesh1412 !

**Pull request merged**